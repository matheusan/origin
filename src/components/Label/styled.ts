import styled from 'styled-components';

export const Wrapper: JSX.Element = styled.div`
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  justify-content: flex-start;
  align-items: stretch;
  align-content: stretch;
`;

export const Label: JSX.Element = styled.label`
  font-size: ${props => props.size}px;
  font-family: ${props => props.theme.fonts.primary};
  color: ${props => props.theme.colors.blueGray900};
  margin-bottom: ${props => (!props.margin ? '0' : '8px')};

  @media only screen and (max-width: 360px) {
    font-size: ${props => props.smallsize || '12'}px;
  }
`;
