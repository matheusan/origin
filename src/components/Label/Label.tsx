import * as React from 'react';

import * as S from './styled';

export interface LabelProps {
  text: string;
  size?: number;
  smallsize?: number;
  margin?: boolean;
}

const Label: React.SFC<LabelProps> = ({
  text,
  size = 14,
  smallsize = null,
  margin = true,
  children
}) => (
  <S.Wrapper>
    {text && (
      <S.Label size={size} smallsize={smallsize} margin={margin}>
        {text}
      </S.Label>
    )}
    {!text ? (
      <S.Label size={size} smallsize={smallsize} margin={margin}>
        {children}
      </S.Label>
    ) : (
      children
    )}
  </S.Wrapper>
);

export default Label;
