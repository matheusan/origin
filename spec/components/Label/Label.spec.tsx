import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import Label from '../../../src/components/Label';

describe('Label', () => {
  let wrapper: ShallowWrapper;

  it('renders', () => {
    wrapper = shallow(<Label>Hello dolly</Label>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render lavel with input', () => {
    wrapper = shallow(
      <Label text="My label">
        <input />
      </Label>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
