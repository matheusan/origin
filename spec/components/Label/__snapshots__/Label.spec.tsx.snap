// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders 1`] = `
<styled.div>
  <styled.label
    margin={true}
    size={14}
    smallsize={null}
  >
    Hello dolly
  </styled.label>
</styled.div>
`;

exports[`Label should render lavel with input 1`] = `
<styled.div>
  <styled.label
    margin={true}
    size={14}
    smallsize={null}
  >
    My label
  </styled.label>
  <input />
</styled.div>
`;
